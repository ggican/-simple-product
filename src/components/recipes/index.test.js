import React from "react";
import Recipes from "./index";
import StyleRecpies from "./index.style";
import RecipesLoading from "./index.loading";

describe("Recipes Component", () => {
    let successElement;

    beforeEach(() => {
        successElement = (
            <Recipes title="Cooking">
                <div test_id="children">children</div>
            </Recipes>
        );
    });

    it("SUCCESS renders and create snapshot  <Recipes />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("SUCCESS renders style and create snapshot  <StyleRecpies />", () => {
        expect(toJson(mount(<StyleRecpies></StyleRecpies>))).toMatchSnapshot();
    });
    it("SUCCESS renders style  and create snapshot <RecipesLoading />", () => {
        expect(
            toJson(mount(<RecipesLoading></RecipesLoading>)),
        ).toMatchSnapshot();
    });

    it("SUCCESS renders and get value title  <Recipes />", () => {
        expect(
            shallow(successElement)
                .find(".recipes__title")
                .text(),
        ).toEqual("Cooking");
    });

    it("SUCCESS renders and get element children  <Recipes />", () => {
        expect(
            shallow(successElement)
                .find(".recipes__list-ingredients")
                .find({ test_id: "children" })
                .text(),
        ).toEqual("children");
    });
    it("SUCCESS renders and get element children  <RecipesLoading />", () => {
        expect(
            shallow(
                <RecipesLoading>
                    <div test_id="children">children</div>
                </RecipesLoading>,
            )
                .find(".recipes__list-ingredients")
                .find({ test_id: "children" })
                .text(),
        ).toEqual("children");
    });
});
