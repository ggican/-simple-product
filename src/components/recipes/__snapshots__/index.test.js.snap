// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes Component SUCCESS renders and create snapshot  <Recipes /> 1`] = `
<Recipes
  title="Cooking"
>
  <styled.div>
    <div
      className="sc-AxjAm ikUmvW"
    >
      <div
        className="recipes"
      >
        <div
          className="recipes__title"
        >
          Cooking
        </div>
        <div
          className="recipes__list-ingredients"
        >
          <div
            test_id="children"
          >
            children
          </div>
        </div>
      </div>
    </div>
  </styled.div>
</Recipes>
`;

exports[`Recipes Component SUCCESS renders style  and create snapshot <RecipesLoading /> 1`] = `
<RecipesLoading>
  <styled.div>
    <div
      className="sc-AxjAm ikUmvW"
    >
      <div
        className="recipes"
      >
        <div
          className="recipes__title"
        >
          <Skeleton
            height="10px"
            marginBottom="7px"
            width="80px"
          >
            <styled.div
              as="div"
              circle={false}
              height="10px"
              marginBottom="7px"
              style={Object {}}
              width="80px"
            >
              <div
                className="sc-AxirZ fokiBL"
                height="10px"
                style={Object {}}
                width="80px"
              />
            </styled.div>
          </Skeleton>
        </div>
        <div
          className="recipes__list-ingredients"
        />
      </div>
    </div>
  </styled.div>
</RecipesLoading>
`;

exports[`Recipes Component SUCCESS renders style and create snapshot  <StyleRecpies /> 1`] = `
<styled.div>
  <div
    className="sc-AxjAm ikUmvW"
  />
</styled.div>
`;
